File: app/src/main/java/com/example/punchpad2/FolderWithNotes.java
package com.stratonotes;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class FolderWithNotes {

    @Embedded
    public FolderEntity folder;

    @Relation(
            parentColumn = "id",
            entityColumn = "folderId"  // <-- must match @ColumnInfo in NoteEntity
    )
    public List<NoteEntity> notes;
}

File: app/src/main/java/com/stratonotes/FolderWithNotes.kt
package com.stratonotes

import androidx.room.Embedded
import androidx.room.Relation
import androidx.room.Ignore

data class FolderWithNotes(
    @Embedded val folder: FolderEntity,

    @Relation(
        parentColumn = "id",
        entityColumn = "folderId"
    )
    val notes: List<NoteEntity>
) {
    @Ignore
    var isExpanded: Boolean = true
}
